skins/oasis/modules/PageHeaderController.class.php

```diff
@@ -8,126 +8,121 @@
-		$this->content_actions = $skinVars['content_actions'];
-		$this->displaytitle = $skinVars['displaytitle']; // if true - don't encode HTML
-		$this->title = $skinVars['title'];
-		$this->subtitle = $skinVars['subtitle'];
+		$this->skinVars = $this->skinTemplate->data;
 	}
 
 	/**
 	 * Use MW core variable to generate action button
+	 *
+	 * @return array $button
 	 */
 	protected function prepareActionButton() {
-
-		global $wgTitle, $wgUser, $wgRequest;
-
-		$isDiff = !is_null( $wgRequest->getVal( 'diff' ) );
-
-		// "Add topic" action
-		if ( isset( $this->content_actions['addsection'] ) ) {
-			// remove on diff pages (RT #72666)
-			if ( $isDiff ) {
-				unset( $this->content_actions['addsection'] );
-			}
-		}
-
-		// action button
-		# print_pre($this->content_actions);
-
-		// handle protected pages (they should have viewsource link and lock icon) - BugId:9494
-		if ( isset( $this->content_actions['viewsource'] ) &&
-			!$wgTitle->isProtected() &&
-			!$wgTitle->isNamespaceProtected( $wgUser ) &&
-			!$wgUser->isLoggedIn() /* VOLDEV-74: logged in users should see the viewsource button, not edit */
+		$wg = $this->wg;
+		$button = [];
+		$actions = $this->skinVars['content_actions'];
+		$isDiff = !is_null( $wg->Request->getVal( 'diff' ) );
+
+		// remove "Add topic" action on diff pages (RT #72666)
+		if ( isset( $actions['addsection'] ) && $isDiff ) {
+			unset( $actions['addsection'] );
+		}
+
+		// handle protected pages (they should have viewsource link and lock icon)
+		// BugId:9494
+		if (
+			isset( $actions['viewsource'] ) &&
+			!$wg->Title->isProtected() &&
+			!$wg->Title->isNamespaceProtected( $wg->User ) &&
+			// VOLDEV-74: logged in users should see the viewsource button, not edit
+			!$wg->User->isLoggedIn()
 		) {
 			// force login to edit page that is not protected
-			$this->content_actions['edit'] = $this->content_actions['viewsource'];
-			$this->content_actions['edit']['text'] = wfMessage( 'edit' )->text();
-			unset( $this->content_actions['viewsource'] );
+			$actions['edit'] = $actions['viewsource'];
+			$actions['edit']['text'] = wfMessage( 'edit' )->escaped();
+			unset( $actions['viewsource'] );
 		}
 
 		// If cascade protected, show viewsource button - BugId:VE-89
-		if ( isset( $this->content_actions['edit'] ) && $wgTitle->isCascadeProtected() ) {
-			$this->content_actions['viewsource'] = $this->content_actions['edit'];
-			$this->content_actions['viewsource']['text'] = wfMessage( 'viewsource' )->text();
-			unset( $this->content_actions['edit'] );
+		if ( isset( $actions['edit'] ) && $wg->Title->isCascadeProtected() ) {
+			$actions['viewsource'] = $actions['edit'];
+			$actions['viewsource']['text'] = wfMessage( 'viewsource' )->escaped();
+			unset( $actions['edit'] );
 		}
 
 		// PvX's rate (RT #76386)
-		if ( isset( $this->content_actions['rate'] ) ) {
-			$this->action = $this->content_actions['rate'];
-			$this->actionName = 'rate';
+		// @todo does this even work? couldn't find it in production
```

Comment:
It's defined in https://github.com/Wikia/app/blob/8a3ebc352c13fd80acce2db2b5fe4a38b99b7eaa/extensions/3rdparty/PvX/Rate.php#L30-30

It's not used anymore. Will create cleanup ticket.